x/epochstorage/keeper/stake_storage.go

```diff
@@ -146,85 +146,90 @@ func (k Keeper) SetStakeStorageCurrent(ctx sdk.Context, chainID string, stakeSto
 	k.SetStakeStorage(ctx, stakeStorage)
 }
 
-func (k Keeper) stakeEntryIndexByAddress(ctx sdk.Context, stakeStorage types.StakeStorage, address sdk.AccAddress) (index uint64, found bool) {
-	// the following finds the address of stakeEntry and returns it
-	entries := stakeStorage.StakeEntries
-	for idx, entry := range entries {
-		entryAddr, err := sdk.AccAddressFromBech32(entry.Address)
-		if err != nil {
+func (k Keeper) GetStakeEntryByAddressFromStorage(ctx sdk.Context, stakeStorage types.StakeStorage, address string) (types.StakeEntry, bool) {
```

Comment:
this can be a method of the stakestorage and not the keeper

Reply:
done. see https://github.com/lavanet/lava/commit/6292ea81beab09c019c0ac964ad0dfb273b9a386